### ISSUE TITLE
Integrate Crossgen2 pack infrastructure into the shared framework tooling in Arcade

### ISSUE THREAD

**Issue by dagood** [dotnet/runtime#1867]
https://github.com/dotnet/runtime/pull/1859 adds some additions to the shared framework infrastructure to make crossgen2 packs:

https://github.com/dotnet/runtime/blob/b8a6d383e231f4d5dc11562f17c2e811bc8d43ee/src/installer/pkg/projects/netcoreapp/pkg/Microsoft.NETCore.App.Crossgen2.pkgproj#L3-L24

These should be ported into dotnet/arcade once the name is decided between Crossgen or Crossgen2, so that shared framework tooling maintenance can happen without as much coordination between repos.

**Comment by dagood** [comment]
Note, this currently produces `.msi` files, but they're not needed and should be disabled when porting for simplicity. The acquisition for crossgen2 will be along the lines of the runtime packs: nuget only. (See https://github.com/dotnet/runtime/issues/1867.)

**Comment by NikolaMilosavljevic** [comment]
This is not a must-have for .NET 5 but it would ensure Crossgen2 generation doesn't get broken.

There's no anticipated sharing of this code.

**Comment by am11** [comment]
Looks like Crossgen2 package has not been published for 5.0:

```sh
$ dotnet --version
5.0.100-rtm.20506.2

$ dotnet new console -n r2rtest; cd r2rtest
The template "Console Application" was created successfully.

Processing post-creation actions...
Running 'dotnet restore' on r2rtest/r2rtest.csproj...
  Determining projects to restore...
  Restored /Users/am11/projects/r2rtest/r2rtest.csproj (in 78 ms).
Restore succeeded.

$ dotnet publish -p:PublishReadyToRun=True -p:PublishReadyToRunUseCrossgen2=True -c Release -r osx-x64
Microsoft (R) Build Engine version 16.8.0-preview-20473-01+250161603 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
/Users/am11/.dotnet/sdk/5.0.100-rtm.20506.2/Sdks/Microsoft.NET.Sdk/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets(63,5): error NETSDK1094: Unable to optimize assemblies for performance: a valid runtime package was not found. Either set the PublishReadyToRun property to false, or use a supported runtime identifier when publishing. [/Users/am11/projects/r2rtest/r2rtest.csproj]
```

**Comment by dagood** [comment]
> Looks like Crossgen2 package has not been published for 5.0:

This issue is about infra quality, and wouldn't be related to this. Can you post a new issue so it can be routed/answered correctly?

I'd also include where you got that SDK and what NuGet feeds you have set up, but that's just intuition, not really knowing likely causes of this error. 😄

**Comment by jkoritzinsky** [comment]
Yes, the tooling was integrated.
